pkg/controllers/status/work_status_controller.go

```diff
@@ -251,10 +251,27 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		// status changes.
 	}
 
+	if !c.isConsistentGenerationState(observedObj) {
+		// When the generation in the resource does not reach a consistent state, skip reflect status and wait for the next update event.
+		klog.Infof("Skip %s(%s/%s) reflect status because generation is inconsistent with current status", observedObj.GetKind(), observedObj.GetNamespace(), observedObj.GetName())
+		return nil
+	}
+
 	klog.Infof("reflecting %s(%s/%s) status to Work(%s/%s)", observedObj.GetKind(), observedObj.GetNamespace(), observedObj.GetName(), workNamespace, workName)
 	return c.reflectStatus(workObject, observedObj)
 }
 
+func (c *WorkStatusController) isConsistentGenerationState(observedObj *unstructured.Unstructured) bool {
+	observedGeneration, ok, err := unstructured.NestedInt64(observedObj.Object, "status", "observedGeneration")
+	if err == nil && ok {
+		// We compare them if `status.observedGeneration` exists.
+		if observedGeneration < observedObj.GetGeneration() {
+			return false
+		}
+	}
```

Comment:
Do you wanna express that the deployment has reached a consistent state if `.status.observedGeneration` is absent?